docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Conversion/MapKeyToVector.cs

```diff
@@ -19,56 +20,62 @@ public static void Example()
 
             // Get a small dataset as an IEnumerable.
             var rawData = new[] {
-                new DataPoint() { Timeframe = 9, Category = 5 },
-                new DataPoint() { Timeframe = 8, Category = 4 },
-                new DataPoint() { Timeframe = 8, Category = 4 },
-                new DataPoint() { Timeframe = 9, Category = 3 },
-                new DataPoint() { Timeframe = 2, Category = 3 },
-                new DataPoint() { Timeframe = 3, Category = 5 }
+                new DataPoint() { Timeframe = 8, PartA=1, PartB=2},
+                new DataPoint() { Timeframe = 7, PartA=2, PartB=1},
+                new DataPoint() { Timeframe = 8, PartA=3, PartB=2},
+                new DataPoint() { Timeframe = 3, PartA=3, PartB=3}
             };
 
             var data = mlContext.Data.LoadFromEnumerable(rawData);
 
-            // Constructs the ML.net pipeline
+            // First transform just maps key type to indicator vector. i.e. it's produces vector of zeros with size of
+            // key cardinality and set 1 to corresponding key's value index in that array.
+            // Next we create vector of keys (same cardinality) and apply two transforms for them.
```

Comment:
>Next we create vector of keys (same cardinality) and apply two transforms for them. [](start = 15, length = 83)

to me the explanation/example sounds confusing:

a) there are 5 transforms in the pipeline, the wording sounds like there are 3 
b)  after concatenation, we do MapValueToKey -- seems out of purview of the API  we are trying to explain  (MapKeyToVector)
c)  need to explain if MapValueToKey  would work on both slots  independently
d) `indexes in array` sounds confusing ..  what would happen if PartA=0 in one of the rows

Reply:
not sure how much better and cleaner it become, but hey, I tried!

Main reason why i'm doing concat + term is because we don't have other way to produce vector of keys, which I wanted to show off for `outputCountVector` parameter.

---
In reply to: [277855878](https://github.com/dotnet/machinelearning/pull/3539#discussion_r277855878) [](ancestors = 277855878)